File: test/fixtures/snippets/ai-style.ts
/**
 * 用户服务模块
 * @author AI Assistant
 * @date 2026-06-28
 */

/**
 * 获取用户信息
 * @param userId - 用户 ID
 * @returns 用户对象
 */
async function fetchUser(userId: string): Promise<Record<string, unknown>> {
  // TODO: 添加缓存层
  try {
    const response = await fetch(`/api/users/${userId}`);
    const data = await response.json();
    return data;
  } catch (error) {
    console.error('Failed to fetch user:', error);
    throw error;
  }
}

/**
 * 更新用户资料
 * @param userId - 用户 ID
 * @param payload - 更新数据
 */
async function updateUser(userId: string, payload: Record<string, unknown>): Promise<void> {
  // TODO: 添加输入校验
  try {
    await fetch(`/api/users/${userId}`, { method: 'PUT', body: JSON.stringify(payload) });
  } catch (error) {
    console.error('Failed to update user:', error);
  }
}

/** 用户数据接口 */
interface UserData {
  id: string;
  name: string;
}

/** 用户仓库接口 */
interface UserRepository {
  findById(id: string): Promise<UserData>;
}

export { fetchUser, updateUser };

File: test/fixtures/snippets/ai-style.java
/**
 * 用户服务实现
 * @author AI Assistant
 */
public class UserServiceImpl implements UserService {

    private String userId;

    /**
     * 获取用户 ID
     * @return 用户 ID
     */
    public String getUserId() {
        return userId;
    }

    /**
     * 设置用户 ID
     * @param userId 用户 ID
     */
    public void setUserId(String userId) {
        this.userId = userId;
    }

    /**
     * 根据 ID 查找用户
     * @param id 用户 ID
     * @return 用户数据
     */
    public UserData findById(String id) {
        try {
            return loadUser(id);
        } catch (Exception e) {
            System.out.println("Error loading user: " + e.getMessage());
        }
        return null;
    }

    private UserData loadUser(String id) {
        return new UserData(id);
    }
}

interface UserService {
    UserData findById(String id);
}

class UserData {
    private String id;

    public UserData(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}
